wagtail/images/views/images.py

```diff
@@ -48,6 +48,7 @@ def get_context_data(self, **kwargs):
             )
             .order_by("-created_at")
             .select_related("collection")
+            .prefetch_related("renditions")
```

Comment:
Question: I believe this will load _all_ renditions into memory, rather than just the ones we need for the index view. For sites with a lot of different images sizes for a given image, this could inflate memory usage quite a lot. Perhaps we could filter this by just the size of image we use in the template?

Reply:
#7225 was taking that approach

Reply:
What about tweaking Karl's PR to work with what has already been merged:
```
class ImageQuerySet(SearchableQuerySetMixin, models.QuerySet):
    def prefetch_renditions(self, relation_name="renditions", filters=None):
        if filters is not None:
            filter_specs = [
                filter.spec if isinstance(filter, Filter) else filter for filter in filters
            ]
            rendition_model = self.model.get_rendition_model()

            return self.prefetch_related(
                models.Prefetch(
                    relation_name,
                    queryset=rendition_model.objects.filter(filter_spec__in=filter_specs),
                )
            )

        return self.prefetch_related(relation_name)
```

Reply:
My personal take is that we should make an effort to ressurect #7225 and get it reviewed/merged. The API is really nice, and perfect for situations like this! I don't think we need to worry about changing it based on #8069, as it makes no sense to use both `prefetch_related()` and `prefetch_renditions()` on the same queryset.

Reply:
I agree with that @ababic. I really think this will be a useful addition. Also with #8590 on the radar, the number of queries can increase drastically if the user wants to show 100 images per page for example.
Also, I was thinking of always using `prefetch_renditions` rather than using both.
@kaedroho what do you think of all of this?

Reply:
@ababic I suspect https://github.com/wagtail/wagtail/pull/7225 could do with a small tweak based on https://github.com/wagtail/wagtail/pull/8069, just to take account of one using `to_attr` with a different name, so improvements like your insertion into the renditions cache for newly generated renditions wouldn't work if using `prefetch_renditions` since it's looking at the wrong name in the cache. That said, should be a quick one!

Reply:
>  just to take account of one using to_attr with a different name

Although there probably is one, I can't immediately think of a nice way to make the various image methods to account different `to_attr` values... so, might just be easiest to always stick with `renditions`.

Reply:
Hmm - I think given we're filtering (and Django recommends `Using to_attr is recommended when filtering down the prefetch result as it is less ambiguous than storing a filtered result in the related manager’s cache`), it might be nicer to check both `renditions` and `prefetched_renditions` (or whatever we end up calling the filtered version), rather than sticking a filtered queryset in `renditions`

Reply:
> it might be nicer to check both renditions and prefetched_renditions (or whatever we end up calling the filtered version), rather than sticking a filtered queryset in renditions

Cool beans! As long as we're not introducing arbitrary `to_attr` support (as suggested by above code snippet), that should be cool.

Reply:
Haha, I agree - I'm not thinking arbitrary support at all, just our short list of full cache + filtered cache names (so likely just 2 possible names).

Reply:
Thanks everyone for looking into this! I've reopened that PR now.

If anyone would like to help finish off this PR, I'd gladly help review it

Reply:
I couldn't work directly on your branch @kaedroho so I made the changes here.